src/inspector_agent.cc

```diff
@@ -477,6 +488,40 @@ void AgentImpl::InstallInspectorOnProcess() {
   v8::Local<v8::Object> inspector = v8::Object::New(env->isolate());
   READONLY_PROPERTY(process, "inspector", inspector);
   env->SetMethod(inspector, "wrapConsoleCall", InspectorWrapConsoleCall);
+  if (options_.wait_for_connect()) {
+    env->SetMethod(inspector, "callAndPauseOnStart", CallAndPauseOnStart);
+  }
+}
+
+// static
+void AgentImpl::CallAndPauseOnStart(
+      const v8::FunctionCallbackInfo<v8::Value>& args) {
+  Environment* env = Environment::GetCurrent(args);
+  v8::HandleScope scope(env->isolate());
+  if (args.Length() < 2) {
+    return env->ThrowError("First argument for a "
+                           "inspector.callAndPauseOnStart call "
+                           "should be a function and the second one should be "
+                           "a thisArg");
+
+  } else if (!args[0]->IsFunction()) {
+    return env->ThrowError("First argument for a "
+                           "inspector.callAndPauseOnStart call "
+                           "should be a function");
+  }
+
+  std::vector<v8::Local<v8::Value>> call_args;
+  for (int i = 2; i < args.Length(); i++) {
+    call_args.push_back(args[i]);
+  }
+
+  env->inspector_agent()->SchedulePauseOnNextStatement("Break on start");
```

Comment:
It breaks on the next JS instruction. E.g. I just noticed that it is not breaking inside the wrapper - it seems to break on the wrapper call (e.g. one step is required to reach the same point it was breaking before). It does not make any difference for the tests as the line number and variables are the same.